src/lib/storage/buffer/frame.hpp

```diff
@@ -0,0 +1,177 @@
+#pragma once
+
+#include <atomic>
+
+#include <bit>
+
+#include "types.hpp"
+
+namespace hyrise {
+
+/**
+ * Frames are the metadata objects for each page. We are using a 64 bit atomic integer to store the (latching) state, NUMA node, dirty flag and the version of the frame.
+ * All operations are atomic. The basic idea and most of the code is based on the SIGMOD'23 paper "Virtual-Memory Assisted Buffer Management" by Leis et al.
+ * 
+ * The frame's upper 16 bits encode the (latching) state (see below). 1 bit is used for the dirty flag. 7 bits are used for the NUMA node. The lower 40 bits are used for the version. 
+ * The version is used to detect concurrent changes to the state of the frame. The version is incremented after unlocking the frame.
```

Comment:
Decremented? (looking at `unlock_shared()`)

How exactly do you determine concurrent changes?

Reply:
The version is incremented for exclusive locks only. For shared locks, the state is decremented and the version stays as is. Concurrent changes are only allowed with exclusive locks

Reply:
> Concurrent changes are only allowed with exclusive locks

You mean any changes are only allowed with exclusive locks?

Can you please adapt the comment to state that the we increment the version only when unlocking from excl. lock?